### PR TITLE
Allow set token with a function and fix a error

### PR DESCRIPTION
**Pull request by Akhail**
This "pull requests" fix an uncaught exception that occurs in a network error, because the response does not have the "status" property. It also allows you to configure the "access token" as a callback function and as a string of course

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,16 +29,16 @@
...
     "axios": "^0.18.0",
     "lodash": "^4.17.11"
   },
   "devDependencies": {
-    "@babel/core": "^7.0.0",
-    "babel-core": "7.0.0-bridge.0",
-    "babel-jest": "^23.4.2",
-    "babel-loader": "^8.0.0",
+    "@babel/core": "^7.1.6",
+    "babel-core": "6.26.3",
```
Comment (KRaymundus) [inline package.json]: Why have you downgraded this?

**Comment by vesper8** [comment]
@Akhail thanks for this! I need this badly!

I have a question though, how do I call this setAuthentication after having already initialized the plugin?

I initialize it as part of my store.js

```
VuexORM.use(VuexORMAxios, {
  database,
  http: {
    baseURL: (process.env.NODE_ENV === 'development' && !process.env.CORDOVA_PLATFORM)
      ? 'https://api.mydomain.dev' : 'https://api.mydomain.com',
    url: '/',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
  },
});
```

then later when my user actually logs in, I want to set the authorization token at that point. How can I access the plugin instance at that point?

And also.. how can I achieve this for now since this PR has not been accepted yet.

Thanks a lot!!

File: package.json (hunk 1 above)
Comment (Akhail) [reply package.json]: Sorry my mistake

**Comment by vesper8** [comment]
thanks for the quick reply @Akhail

I just tried out your fork and unfortunately it doesn't work out for me.. I also don't understand how this could work for you. How can you access the user's token from within that callback? You can't even access 'this'.. let alone the $auth global that has access to the token

I use @websanova/vue-auth to handle my jwt login/access and I tried doing this:

```
    access_token: () => {
      if (this.$auth.check()) {
        return this.$auth.token();
      }
      return '';
    },
```

It tells me that $auth is not accessible. this is not accessible either.

In your mock example you refer to a variable "login" but how could you even have access to that in the scope of your store.js ?

**Comment by vesper8** [comment]
Oh my.. I kept reading and trying things and I guess I got lucky.. this works!

```
    access_token() {
      if (Vue.auth.check()) {
        return Vue.auth.token();
      }
      return '';
    },
```

I also discovered this awesome package here https://github.com/ds300/patch-package which lets you create a patch super easily so you don't have to rely on forks when fixing/adding features such as in this case

Anyway your method works awesome and is much needed @Akhail ! Thanks a lot for helping put me on the right patch and for this PR!

@kiaking @KaniRobinson please please merge this PR!!

**Comment by fowkswe** [comment]
For those wondering how to get the token from something like nuxt auth-module, I'm getting it from localStorage:

```
    access_token() {
      return window.localStorage['auth._token.local'].substr('Bearer '.length);
    }
```

Thanks to @go4cas in the Slack channel for this suggestion.

**Comment by twnaing** [comment]
@fowkswe, I couldn't access `window`.

@vesper8 , I am using this with nuxtjs and `Vue.auth` also doesn't work.

@Akhail , how can I check nuxtjs auth-module token?

From documentation, I should be able to access it using

```js
localStorage.getItem("token") 
// or
$store.state.auth.getToken('local')
```

**Comment by twnaing** [comment]
For Nuxtjs, I have to create plugin and use that plugin (in `nuxt.config.js`) with `mode: client` and I can access auth using @fowkswe code.

**Pull request merged**